histoqc/SaveModule.py

```diff
@@ -62,6 +72,11 @@ def saveAssociatedImage(s, key:str, dim:int):
     associated_img = np.asarray(associated_img)[:, :, 0:3]
     io.imsave(f"{s['outdir']}{os.sep}{s['filename']}_{key}.png", associated_img)
 
+def saveLabel(s, params):
```

Comment:
Actually, we have a single function named **saveAssociatedImage** with the string parameter. The **saveLabel** and **saveMacro** functions are restrictions of using **saveAssociatedImage** since wsidicom and openslide image and APIs are heterogeneous. In openslide, **associated_images** property is _**Images, such as label or macro images, which are associated with this slide. This is a Mapping from the image name to RGBA Image.**_ We can directly get the associated image by the key such as **openslide.associated_images["macro"]**. However, in wsidicom, there is no associated image property. The wsidicom only provides 2 functions to get associated images - **WsiDicom.read_label()** and **WsiDicom.read_overview()**. 
Saving label and macro functions should be the most frequently used functions. This is why I create these 2 functions individually. Open to any better proposal. ;)

Reply:
i think you can remove both of these functions, and have them be keywords used directly in "saveAssociatedImage" via the config file?

you basically already have this, except instead of having "key" come in as a method-level parameter, you read it from the config file?